source/winAPI/sessionTracking.py

```diff
@@ -155,74 +122,37 @@ def isWindowsLocked() -> bool:
 	Not to be confused with the Windows sign-in screen, a secure screen.
 	"""
 	from core import _TrackNVDAInitialization
+	from winAPI.sessionTracking import _isWindowsLocked_checkViaSessionQuery
+	from systemUtils import _isSecureDesktop
 	if not _TrackNVDAInitialization.isInitializationComplete():
-		# During NVDA initialization,
-		# core._initializeObjectCaches needs to cache the desktop object,
-		# regardless of lock state.
-		# Security checks may cause the desktop object to not be set if NVDA starts on the lock screen.
-		# As such, during initialization, NVDA should behave as if Windows is unlocked.
 		return False
 	if _isSecureDesktop():
-		# If this is the Secure Desktop,
-		# we are in secure mode and on a secure screen,
-		# e.g. on the sign-in screen.
-		# _isSecureDesktop may also return True on the lock screen before a user has signed in.
-
-		# For more information, refer to devDocs/technicalDesignOverview.md 'Logging in secure mode'
-		# and the following userGuide sections:
-		# - SystemWideParameters (information on the serviceDebug parameter)
-		# - SecureMode and SecureScreens
 		return False
-	lockStateTracked = _hasLockStateBeenTracked()
-	if lockStateTracked:
-		return WindowsTrackedSession.SESSION_LOCK in _currentSessionStates
+	global _initializationLockState
+	if _lockStateTracker is not None:
+		_isWindowsLocked = _lockStateTracker.isWindowsLocked
 	else:
-		_recordLockStateTrackingFailure()  # Report error repeatedly, attention is required.
-		##
-		# For security it would be best to treat unknown as locked.
-		# However, this would make NVDA unusable.
-		# Instead, the user should be warned via UI, and allowed to mitigate the impact themselves.
-		# See usage of L{hasLockStateBeenUnknown}.
-		return False  # return False, indicating unlocked, to allow NVDA to be used
-
-
-def _setInitialWindowLockState() -> None:
-	"""
-	Ensure that session tracking state is initialized.
-	If NVDA has started on a lockScreen, it needs to be aware of this.
-	As NVDA has already registered for session tracking notifications,
-	a lock is used to prevent conflicts.
-	"""
-	with _updateSessionStateLock:
-		lockStateTracked = _hasLockStateBeenTracked()
-		if lockStateTracked:
-			log.debugWarning(
-				"Initial state already set."
-				" NVDA may have received a session change notification before initialising"
-			)
-		# Fall back to explicit query
-		try:
-			isLocked = _isWindowsLocked_checkViaSessionQuery()
-			_currentSessionStates.add(
-				WindowsTrackedSession.SESSION_LOCK
-				if isLocked
-				else WindowsTrackedSession.SESSION_UNLOCK
-			)
-		except RuntimeError as error:
-			_recordLockStateTrackingFailure(error)
+		if _initializationLockState is None:
+			_initializationLockState = _isWindowsLocked_checkViaSessionQuery()
+		_isWindowsLocked = _initializationLockState

```

Comment:
`_initializationLockState` is used here. The idea is that we query the session state only once during initialization.

Reply:
This is actually redundant due to the `_TrackNVDAInitialization.isInitializationComplete()` check. I will remove `_initializationLockState` handling.